# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1510258062 -3600
#      Thu Nov 09 21:07:42 2017 +0100
# Node ID 629858e122f148a85c7fed295b7e03df83022f37
# Parent  7d1561c9ee4866656f741ce83b1d97100e230f42
do not extend selection when indenting/commenting in editor (bug #45610)

* file-editor-tab.cc (do_indent_selected_text, do_comment_selected_text):
  extend selction to line lenght - 1 only

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -1402,17 +1402,17 @@ file_editor_tab::do_indent_selected_text
         {
           if (indent)
             _edit_area->indent (i);
           else
             _edit_area->unindent (i);
         }
       //set selection on (un)indented section
       _edit_area->setSelection (lineFrom, 0, lineTo,
-                                _edit_area->text (lineTo).length ());
+                                _edit_area->text (lineTo).length ()-1);
     }
   else
     {
       int cpline, col;
       _edit_area->getCursorPosition (&cpline, &col);
       if (indent)
         _edit_area->indent (cpline);
       else
@@ -1447,17 +1447,17 @@ file_editor_tab::do_comment_selected_tex
                 {
                   _edit_area->setSelection (i, 0, i, comment_str.length ());
                   _edit_area->removeSelectedText ();
                 }
             }
         }
       //set selection on (un)commented section
       _edit_area->setSelection (lineFrom, 0, lineTo,
-                                _edit_area->text (lineTo).length ());
+                                _edit_area->text (lineTo).length ()-1);
     }
   else
     {
       int cpline, col;
       _edit_area->getCursorPosition (&cpline, &col);
       if (comment)
         _edit_area->insertAt (comment_str, cpline, 0);
       else
